beacon-chain/rpc/eth/light-client/handlers.go

```diff
@@ -117,108 +115,29 @@ func (s *Server) GetLightClientUpdatesByRange(w http.ResponseWriter, req *http.R
 		endPeriod = maxSlot / slotsPerPeriod
 	}
 
-	// Populate updates
-	var updates []*structs.LightClientUpdateResponse
-	for period := startPeriod; period <= endPeriod; period++ {
-		// Get the last known state of the period,
-		//    1. We wish the block has a parent in the same period if possible
-		//	  2. We wish the block has a state in the same period
-		lastSlotInPeriod := period*slotsPerPeriod + slotsPerPeriod - 1
-		if lastSlotInPeriod > maxSlot {
-			lastSlotInPeriod = maxSlot
-		}
-		firstSlotInPeriod := period * slotsPerPeriod
-
-		// Let's not use the first slot in the period, otherwise the attested header will be in previous period
-		firstSlotInPeriod++
-
-		var state state.BeaconState
-		var block interfaces.ReadOnlySignedBeaconBlock
-		for slot := lastSlotInPeriod; slot >= firstSlotInPeriod; slot-- {
-			state, err = s.Stater.StateBySlot(ctx, types.Slot(slot))
-			if err != nil {
-				continue
-			}
-
-			// Get the block
-			latestBlockHeader := state.LatestBlockHeader()
-			latestStateRoot, err := state.HashTreeRoot(ctx)
-			if err != nil {
-				continue
-			}
-			latestBlockHeader.StateRoot = latestStateRoot[:]
-			blockRoot, err := latestBlockHeader.HashTreeRoot()
-			if err != nil {
-				continue
-			}
-
-			block, err = s.Blocker.Block(ctx, blockRoot[:])
-			if err != nil || block == nil {
-				continue
-			}
-
-			syncAggregate, err := block.Block().Body().SyncAggregate()
-			if err != nil || syncAggregate == nil {
-				continue
-			}
-
-			if syncAggregate.SyncCommitteeBits.Count()*3 < config.SyncCommitteeSize*2 {
-				// Not enough votes
-				continue
-			}
-
-			break
-		}
+	// get updates
+	updatesMap, err := s.BeaconDB.LightClientUpdates(ctx, startPeriod, endPeriod)
+	if err != nil {
+		httputil.HandleError(w, "Could not get light client updates from DB: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
 
-		if block == nil {
-			// No valid block found for the period
-			continue
-		}
+	updates := make([]*structs.LightClientUpdateResponse, 0, len(updatesMap))
 
-		// Get attested state
-		attestedRoot := block.Block().ParentRoot()
-		attestedBlock, err := s.Blocker.Block(ctx, attestedRoot[:])
-		if err != nil || attestedBlock == nil {
-			continue
+	for i := startPeriod; i <= endPeriod; i = i + 1 {
```

Comment:
```suggestion
	for i := startPeriod; i <= endPeriod;  i++ {
```
Just less verbose rest upto your choice